src/views/Alteration.vue

```diff
@@ -256,6 +248,16 @@ export default class Alteration extends Mixins(
     return ''
   }
 
+  /** The entity specific resource file for an Alteration filing. */
+  private get alterationResources (): any {
+    const resources = AlterationResources.find(x => x.entityType === this.getEntityType)
+    if (!resources) {
+      // go to catch()
+      throw new Error(`Invalid Alteration resources entity type = ${this.getEntityType}`)
```

Comment:
This seems to work OK because this getter is only use by code (not by template).

Personally, I'd be tempted to just return the result from `find` and let the caller deal with possible null/undefined response. That would move the error-checking logic entirely into the caller, which would be more clear. Currently some of that logic is hidden here -- the caller would probably not know that this getter could throw an exception. (In some cases we explicitly say that some code might throw an error, just to show that error handling is distributed between caller and callee.)

Reply:
For sure, i see what you mean. 
I'm not sure how to refactor. 
Seems verbose to reference and check the getter before it's used it imo but i'm open to other ideas!

Reply:
Have the getter just return the data you're looking for (line 253).

Move the check on lines 254-257 to the caller. Then all the checks and try-catch are in the same place.

Reply:
Updated, the resources actions were already wrapped in a try catch, so added a check throw the error if required. 
Same for `Change.vue`